### PR TITLE
Protect against prototype pollution in import action

### LINKED ISSUE #7089: json should be sanitized on import
#### Summary
Prevent users from polluting the object's prototype.

#### Expected vs Current Behavior
Should not be permitted to pollute prototype using import function.
Should not be permitted to pollute prototype by modifying local storage.

#### Testing Instructions (import)
1. open the browser console
2. right click on and export an object from the tree
3. use a text editor to modify the exported json to include a "__proto__" key (or use attached)
4. right click a container in the tree and select import
5. import the exported json
6. Acceptance - Observe the import works and the object is imported into the tree
7. Acceptance - observe there are no console errors
8. Acceptance - examine the object (see below) in the console and observe the prototype of the imported domain object is not changed

To examine the object
1. get the object id from the url `{HOST}/#/browse/mine/{ID}?{PARAM_STRING}
10. in the console type `openmct.objects.get('{ID}'), where {ID} is the id from step 1
11. expand the object as needed

#### Testing Instructions (localstorage)
1. must have the local storage plugin enabled (local development will enable)
2. save an object
3. open the browser console
4. click on application->local storage->HOST
5. locate the key `mct`
6. modify the saved object to have a "__proto__" key ("__proto__":{"foo":"bar"}","identifier"....)
7. refresh the browser
9. edit and save the object in the UI
10. Acceptance - Observe the object still is available in the tree
11. Acceptance - observe the Value of the local storage string no longer contains "__proto__"

#### Environment

* Open MCT Version: 
* Deployment Type: 
* OS:
* Browser:

#### Impact Check List

- [ ] Data loss or misrepresented data?
- [ ] Regression? Did this used to work or has it always been broken?
- [ ] Is there a workaround available?
- [x] Does this impact a critical component?
- [ ] Is this just a visual bug with no functional impact?
- [ ] Does this block the execution of e2e tests?
- [ ] Does this have an impact on Performance?

#### Additional Information
```{"openmct":{"c28d230d-e909-4a3e-9840-d9ef469dda70":{"identifier":{"key":"c28d230d-e909-4a3e-9840-d9ef469dda70","namespace":""},"name":"Unnamed Overlay Plot","type":"telemetry.plot.overlay","composition":[],"configuration":{"series":[]},"modified":1695837546833,"location":"mine","created":1695837546833,"persisted":1695837546833,"__proto__":{"toString":"foobar"}}},"rootId":"c28d230d-e909-4a3e-9840-d9ef469dda70"}```

### PR DESCRIPTION
**Pull request by davetsay**
Closes #7089

### Describe your changes:
remove `__proto__` when parsing JSON from import action

### All Submissions:

* [x] Have you followed the guidelines in our [Contributing document](https://github.com/nasa/openmct/blob/master/CONTRIBUTING.md)?
* [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/nasa/openmct/pulls) for the same update/change?
* [x] Is this change backwards compatible? For example, developers won't need to change how they are calling the API or how they've extended core plugins such as Tables or Plots.

### Author Checklist

* [x] Changes address original issue?
* [x] Tests included and/or updated with changes?
* [x] Command line build passes?
* [x] Has this been smoke tested?
* [x] Testing instructions included in associated issue OR is this a dependency/testcase change?

### Reviewer Checklist

* [ ] Changes appear to address issue?
* [ ] Reviewer has tested changes by following the provided instructions?
* [ ] Changes appear not to be breaking changes?
* [ ] Appropriate automated tests included?
* [ ] Code style and in-line documentation are appropriate?
* [ ] Has associated issue been labelled unverified? (only applicable if this PR closes the issue)
* [ ] Has associated issue been labelled bug? (only applicable if this PR is for a bug fix)

### CODE REVIEW

**Comment by unlikelyzero** [comment]
Can you add a functional and negative test scenario for this in the e2e suite

File: src/plugins/localStorage/LocalStorageObjectProvider.js [hunk 1]
```diff
@@ -83,7 +85,7 @@ export default class LocalStorageObjectProvider {
    * @private
    */
   getSpaceAsObject() {
-    return JSON.parse(this.getSpace());
+    return JSON.parse(this.getSpace(), filter__proto__);
```
Comment (shefalijoshi) [inline src/plugins/localStorage/LocalStorageObjectProvider.js:88]: Can we use Vue's `toRaw`utility here?
Something like:
`return JSON.parse(toRaw(this.getSpace())`

File: src/plugins/importFromJSONAction/ImportFromJSONAction.js [hunk 2]
```diff
@@ -71,8 +72,10 @@ export default class ImportAsJSONAction {
 
   onSave(object, changes) {
     const selectFile = changes.selectFile;
-    const objectTree = selectFile.body;
-    this._importObjectTree(object, JSON.parse(objectTree));
+    const jsonTree = selectFile.body;
+    const objectTree = JSON.parse(jsonTree, filter__proto__);
```
Comment (shefalijoshi) [inline src/plugins/importFromJSONAction/ImportFromJSONAction.js:76]: Can we use Vue's toRawutility here?
Something like:
return JSON.parse(toRaw(jsonTree)

**Review by shefalijoshi — COMMENTED** [review]
Curious if we can use Vue's utility method here instead?

**Comment by davetsay** [comment]
> Curious if we can use Vue's utility method here instead?

I think we should try to keep view/Vue code out of the logic paths if possible.

**Review by shefalijoshi — APPROVED** [review]
Looks fine!

**Review by ozyx — CHANGES_REQUESTED** [review]
Looks like there's a new memory leak? thus the failing perf test

**Review by ozyx — APPROVED** [review]

**Comment by davetsay** [comment]
> Looks like there's a new memory leak? thus the failing perf test

I re-ran the tests and perf passes. I don't think this introduced a leak.

**Pull request merged**